### PR TITLE
Fix broadcast

### LINKED ISSUE #532: Cannot broadcast arrays with shapes (..., 1, ...) and (..., 0, ...)
```
>>> cp.__version__
'2.0.0rc1'
>>> cp.broadcast(cp.ones((1,)), cp.ones((0,))).shape
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "cupy/core/core.pyx", line 2260, in cupy.core.core.broadcast.__init__
ValueError: Broadcasting failed
```
is inconsistent with
```
>>> np.broadcast(np.ones((1,)), np.ones((0,))).shape
(0,)
```

### PR DESCRIPTION
**Pull request by niboshi**
Fixes #532

### CODE REVIEW

File: cupy/core/core.pyx [hunk 1]
```diff
@@ -2257,7 +2266,7 @@ cdef class broadcast:
                 a_sh = a._shape[a_ndim - i - 1]
                 if a_sh == r_shape[i]:
                     r_strides[i] = a._strides[a_ndim - i - 1]
-                elif a_sh != 1:
+                elif a_sh > 1:
```
Comment (toslunar) [inline cupy/core/core.pyx]: Why is the change needed?

Comment (niboshi) [reply cupy/core/core.pyx]: Because `a_sh` will be `0` if all the corresponding dimensions of the arrays are either 0 or 1.

Comment (toslunar) [reply cupy/core/core.pyx]: I got your intension.

I found `python -m unittest tests/cupy_tests/manipulation_tests/test_dims.py` passes with `elif a_sh != 0:`, too.  If `a_sh == 0` then it seems `a_sh == r_shape[i]` holds.

Comment (niboshi) [reply cupy/core/core.pyx]: It seems you're right. Thank you for pointing out.

**Comment by niboshi** [comment]
Fixed to reflect the review.

**Comment by takagi** [comment]
LGTM!

**Pull request merged**